parachains/pallets/collective-content/src/lib.rs

```diff
@@ -0,0 +1,276 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Managed Collective Content Pallet
+//!
+//! The pallet provides the functionality to store different types of content. This would typically
+//! be used by an on-chain collective, such as the Polkadot Alliance or Ambassador Program.
+//!
+//! The pallet stores content as a [Cid], which should correspond to some off-chain hosting service,
+//! such as IPFS, and contain any type of data. Each type of content has its own origin from which
+//! it can be managed. The origins are configurable in the runtime. Storing content does not require
+//! a deposit, as it is expected to be managed by a trusted collective.
+//!
+//! Content types:
+//!
+//! - Collective [charter](pallet::Charter): A single document (`Cid`) managed by
+//!   [CharterOrigin](pallet::Config::CharterOrigin).
+//! - Collective [announcements](pallet::Announcements): A list of announcements managed by
+//!   [AnnouncementOrigin](pallet::Config::AnnouncementOrigin).
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+use frame_support::{traits::schedule::DispatchTime, BoundedVec};
+use sp_core::ConstU32;
+use sp_std::prelude::*;
+
+/// IPFS compatible CID.
+// worst case 2 bytes base and codec, 2 bytes hash type and size, 64 bytes hash digest.
+pub type Cid = BoundedVec<u8, ConstU32<68>>;
+
+/// The block number type of [frame_system::Config].
+pub type BlockNumberFor<T> = <T as frame_system::Config>::BlockNumber;
+
+/// [DispatchTime] of [frame_system::Config].
+pub type DispatchTimeFor<T> = DispatchTime<BlockNumberFor<T>>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
+
+	/// The module configuration trait.
+	#[pallet::config]
+	pub trait Config<I: 'static = ()>: frame_system::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self, I>>
+			+ IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The origin to control the collective announcements.
+		type AnnouncementOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The origin to control the collective charter.
+		type CharterOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The maximum number of announcements.
+		///
+		/// NOTE: Benchmarks need to be re-run if this changes.
+		#[pallet::constant]
+		type MaxAnnouncementsCount: Get<u32>;
```

Comment:
I don't think you need `Count` here, it reads fine without it:
```suggestion
		type MaxAnnouncements: Get<u32>;
```

Reply:
If you check substrate pallets, this is usual suffix for such counts